src/vsphere_cpi/lib/cloud/vsphere/attach_tag_to_vm.rb

```diff
@@ -0,0 +1,243 @@
+require 'cloud/vsphere/logger'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+
+module  VSphereCloud
+  module TaggingTag
+
+    class VCenterBadEntityError < StandardError
+      def initialize(name, type)
+        @name = name
+        @type = type
+      end
+
+      def to_s
+        "Unable to locate #{@type} with name '#{@name}' on vCenter, skip attaching this #{@type}."
+      end
+    end
+
+    class VCenterNoTagFoundError < StandardError
```

Comment:
Can the names be consistent Tag has `VCenterNoTagFoundError ` and category has `VCenterNoCategoryExitsError` ?

Reply:
Done.